frontend/src/views/consumer/event-listing/index.js

```diff
@@ -1,51 +1,139 @@
+import Typography from '../../../components/typography';
+import getMonth from 'date-fns/get_month';
 import { values } from 'mobx';
+import posed, { PoseGroup } from 'react-pose';
+import Button from '../../../components/button';
+
+const ContainerAnimation = posed.div({
+  enter: {
+    delay: 600,
+    scale: 1,
+    opacity: 1,
+    staggerChildren: 300,
+  },
+  exit: {
+    delay: 300,
+    scale: 0,
+    opacity: 0,
+    staggerChildren: 200,
+  },
+  preEnter: {
+    scale: 0,
+    y: '-5%',
+  },
+});
+const CardContainers = styled(ContainerAnimation)``;
+const EmptyStateContainer = styled(ContainerAnimation)`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  color: rgba(0, 0, 0, 0.5);
+  font-size: 2rem;
+
+  svg {
+    fill: ${props => props.theme.palette.primaryDeep};
+    width: 5rem;
+  }
+  & > * {
+    margin: 1rem;
+  }
+`;
 
 const Wrapper = styled.div`
   display: flex;
   flex-direction: column;
   height: 100%;
   width: 100%;
   overflow: auto;
-  padding: 2rem 1rem;
+  padding: 1rem;
+  background-image: linear-gradient(
+    to top,
+    #d5d4d0 0%,
+    #d5d4d0 1%,
+    #eeeeec 31%,
+    #efeeec 75%,
+    #e9e9e7 100%
+  );
 
   & > * {
     flex-shrink: 0;
     flex-grow: 0;
   }
 `;
+const filterByAgeGroup = age => events => {
+  return events.filter(event => event.ageGroupLimit === age);
+};
+const filterByArea = area => events => {
+  return events.filter(event => event.area === area);
+};
+const filterByDate = month => events =>
+  events.filter(event => getMonth(event.date) === month);
 
-export default connect('store')(
-  class EventListing extends Component {
-    render() {
-      const { events, selectedEvent } = this.props.store;
-      return (
-        <Wrapper>
-          {!this.props.store.isEmpty ? (
-            values(events).map(event => (
-              <EventCard
-                expandable
-                active={selectedEvent && selectedEvent.id == event.id}
-                style={{ marginBottom: '0.7rem' }}
-                key={event.id}
-                event={event}
-                onSelect={() => this.props.store.selectEvent(event.id)}
-                onDeselect={this.props.store.deselectEvent}
-              >
-                <EventDetail event={event} />
-                <EventBooking event={event} />
-              </EventCard>
-            ))
-          ) : (
-            <h1>Nothing to see here</h1>
-          )}
-        </Wrapper>
-      );
-    }
-  },
+const filterByEventType = eventType => events =>
+  events.filter(event => event.eventType === eventType);
+
+export default withTheme(
+  connect('store')(
+    class EventListing extends Component {
+      render() {
+        const { events, selectedEvent, filters } = this.props.store;
+        const { ageGroupLimit, area, eventType, date } = filters;
+
+        const displayableEvents = pipeable(values(events)).pipe(
```

Comment:
Are you sure it's a good idea to do the filtering inside the render method?

Reply:
Filtered event is just a derived data. I think its a fitting place. otherwise stale data would arise